extensions/sql-database-projects/src/common/utils.ts

```diff
@@ -95,3 +96,20 @@ export function getSafeNonWindowsPath(filePath: string): string {
 	filePath = filePath.split('\\').join('/').split('"').join('');
 	return '"' + filePath + '"';
 }
+
+/**
+ * Read SQLCMD variables from xmlDoc and return them
+ * @param xmlDoc xml doc to read SQLCMD variables from. Format must be the same that sqlproj and publish profiles use
+ */
+export function readSqlCmdVariables(xmlDoc: any): Record<string, string> {
```

Comment:
Fine for preview.  I feel like deployment profile stuff should eventually (sooner rather than later) be moved to its own class, though, rather than with the generic utils.

Reply:
Sounds good. Will move deployment profile stuff in another PR